api/utils/graphs.ts

```diff
@@ -2,6 +2,31 @@ import { JSONObjectStringifier, StringifiedMap } from "df/common/strings/stringi
 import { IActionProto } from "df/core/session";
 import { dataform } from "df/protos/ts";
 
+const types = ["dataset", "assertion", "operation", "declaration"] as const;
+export type Type = typeof types[number];
+
+export interface IAction {
+  name: string;
+  type: Type;
```

Comment:
as in the other PR, i still don't think we need this... `instanceof` guards should be enough.

TBH i think we should get rid of this interface altogether and just have an `allActions` function that concatenates all of the actions and returns the result.